addons/website_blog/controllers/main.py

```diff
@@ -170,6 +170,19 @@ def tags_list(tag_ids, current_tag):
         response = request.website.render("website_blog.blog_post_short", values)
         return response
 
+    @http.route(['/blog/<model("blog.blog"):blog>/<any("atom","rss"):feed_type>'], type='http', auth="public")
+    def blog_feed(self, blog, feed_type, **post):
+        limit = int(post.get('limit', 15))
+        cr, uid, context = request.cr, request.uid, request.context
+        base_url = request.registry['ir.config_parameter'].get_param(cr, uid, 'web.base.url')
+        blog_post_obj = request.registry['blog.post']
+        blog_post_ids = blog_post_obj.search(cr, uid, [('blog_id', '=', blog.id)], order="write_date desc", limit=limit, context=context)
+        blog_posts = blog_post_obj.browse(cr, uid, blog_post_ids, context=context)
```

Comment:
Instead of a browse, do a read (with the field you need). Before rendering, prepare the data : for date field, convert string to datetime, set the timezone as UTC, a convert it to string with correct format (iso for atom, other for rss) in the dict of value to give to render the view.